docs/rule/groovy_library.soy

```diff
@@ -0,0 +1,168 @@
+  <code>.java</code>, cross compilation using the jdk found in <code>JAVA_HOME</code> will occur.
+  {/param}
+{/call}
+
+{call buck.arg}
+  {param name: 'resources' /}
+  {param default : '[]' /}
+  {param desc}
+  This is the same as in <a href="{ROOT}rule/java_library.html"><code>java_library</code></a>.
+  {/param}
+{/call}
+
+{call buck.arg}
+  {param name: 'deps' /}
+  {param default : '[]' /}
+  {param desc}
+  Rules (usually other <code>groovy_library</code> or <code>java_library</code> rules) that are
+  used to generate the classpath required to compile this <code>groovy_library</code>.
+
+  This is the same as in <a href="{ROOT}rule/java_library.html"><code>java_library</code></a>.
+  {/param}
+{/call}
+
+{call buck.arg}
+  {param name: 'exported_deps' /}
+  {param default : '[]' /}
+  {param desc}
+  Other <code>groovy_library</code> and <code>java_library</code> rules that depend
+  on this rule will also include its <code>exported_deps</code> in their classpaths.
+
+  This is the same as in <a href="{ROOT}rule/java_library.html"><code>java_library</code></a>.
+  {/param}
+{/call}
+
+{call buck.arg}
+  {param name: 'provided_deps' /}
+  {param default : '[]' /}
+  {param desc}
+  This is the same as in <a href="{ROOT}rule/java_library.html"><code>java_library</code></a>.
+  {/param}
+{/call}
+
+{call buck.arg}
+  {param name: 'extra_groovyc_arguments' /}
+  {param default : '[]' /}
+  {param desc}
+  List of additional arguments to pass into the Groovy compiler.
+  {/param}
+{/call}
+
+{call buck.arg}
+  {param name: 'source' /}
+  {param default : '<global value>' /}
+  {param desc}
+  Only used during cross compilation.
+
+  This is the same as in <a href="{ROOT}rule/java_library.html"><code>java_library</code></a>.
+  {/param}
+{/call}
+
+{call buck.arg}
+  {param name: 'target' /}
+  {param default : '<global value>' /}
+  {param desc}
+  Only used during cross compilation.
+
+  This is the same as in <a href="{ROOT}rule/java_library.html"><code>java_library</code></a>.
+  {/param}
+{/call}
+
+{call buck.arg}
+  {param name: 'java_version' /}
+  {param default: '<global value>' /}
+  {param desc}
+  Only used during cross compilation.
+
+  This is the same as in <a href="{ROOT}rule/java_library.html"><code>java_library</code></a>.
+  {/param} 
+{/call}
+
+{call buck.arg}
+  {param name: 'extra_arguments' /}
+  {param default : '[]' /}
+  {param desc}
+  Only used during cross compilation.
+
+  This is the same as in <a href="{ROOT}rule/java_library.html"><code>java_library</code></a>.
+  {/param}
+{/call}
+
+{call buck.visibility_arg /}
+
+{/param} // close args
+
+{param examples}
+
+{literal}<pre class="prettyprint lang-py">
+# A rule that compiles a single .groovy file.
+groovy_library(
+  name = 'example',
+  srcs = ['MySourceFile.groovy'],
+)
+
+# A rule that compiles all of the .groovy files under the directory in
+# which the rule is defined using glob()
```

Comment:
It's better to do `<p>` before each example and have each one in its own box, IMO